services/QuillLMS/app/workers/archive_classroom_units_worker.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class ArchiveClassroomUnitsWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: SidekiqQueue::HIGH_MEMORY
+
+  def perform(classroom_ids)
+    failed_classrooms = []
+
+    classroom_ids.each do |classroom_id|
+      classroom = Classroom.unscoped.find(classroom_id)
+      classroom.hide_all_classroom_units
+    rescue => e
+      Rails.logger.error "Failed to archive classroom units for classroom #{classroom_id}: #{e.message}"
+      Rails.logger.error e.backtrace.join("\n")
+      failed_classrooms << { id: classroom_id, error: e.message }
+    end
+
+    return unless failed_classrooms.any?
+
+    raise "Failed to process #{failed_classrooms.size} classrooms: #{failed_classrooms.map { |f| "#{f[:id]} (#{f[:error]})" }.join(', ')}"
```

Comment:
To address Thomas's concern a different way, you could alternatively recursively call the worker here wit the failed classroom ids, instead of `raise`